src/components/People.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useContext } from 'react';
+import cn from 'classnames';
+import PeopleContext from '../contexts/PeopleContext';
+
+export const People = React.memo(() => {
+  const {
+    input,
+    setInput,
+    applyQueryWithDebounce,
+    filteredPeople,
+    handleClick,
+  } = useContext(PeopleContext);
+
+  const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const userInput = event.target.value;
+
+    applyQueryWithDebounce(userInput);
```

Comment:
You should pass the 'userInput' to the 'setInput' function before calling 'applyQueryWithDebounce' to ensure the input state is updated before debouncing.